File: canteen-service/database/author.ts
import { Document, ObjectId } from "mongodb";
import { DatabaseName, getClient } from "./client";

const Authors = "Authors";

export interface AuthorEntity extends Document {
    Name: string
    Title: string
    Region: string
    Address: string
    CreatedAt: Date,
    UpdatedAt: Date
}

export interface Author {
    id: string,
    name: string,
    title: string,
    region: string,
    address: string,
}

export async function getAuthors(): Promise<Author[]> {
    const client = await getClient();
    const collection = client.db(DatabaseName).collection<AuthorEntity>(Authors);
    const authors = await collection.find().toArray();
    await client.close();
    return authors.map(a => {
        return {
            id: a._id.toString(),
            name: a.Name,
            title: a.Title,
            region: a.Region,
            address: a.Address
        };
    });
}

File: factory-service/models/author.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type AuthorEntity struct {
	Id        primitive.ObjectID `bson:"_id"`
	Name      string             `bson:"name"`
	Title     string             `bson:"title"`
	Region    string             `bson:"region"`
	Address   string             `bson:"address"`
	CreatedAt string             `bson:"createdAt"`
	Timestamp string             `bson:"timestamp"`
}
